deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -172,6 +175,18 @@ def copy_sim_slot_infrastructure(self, slotno: int) -> None:
 
             run("""cp -r {}/* {}/""".format(remote_sim_rsync_dir, remote_sim_dir), shell=True)
 
+    def extract_driver_tarball(self, slotno: int) -> None:
+        """ extract tarball that already exists on the remote node. """
+        if self.instance_assigned_simulations():
+            assert slotno < len(self.parent_node.sim_slots)
+            serv = self.parent_node.sim_slots[slotno]
+
+            remote_sim_dir = self.get_remote_sim_dir_for_slot(slotno)
+
+            options = "-zxf"
+            cmd = f"cd {remote_sim_dir} && tar {options} {serv.get_tar_name()}"
```

Comment:
So for `cd` fabric provides a context manager you can use instead. And then, nit: I'd then just put the run command inline 
i.e., `run(f"tar {options} {serv.get_tar_name()")` as it is done generally throughout the codebase. I find it more readable because of the consistency, but i'll let you decide.